guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+
+"""
+KerasCV offers a set of APIs to train LIDAR-based 3D object detection models,
+including dataloading, augmentation, model training, and metric evaluation.
+problems. These APIs were designed and implemented in partnership with Waymo.
+
+In this guide, we'll take KerasCV's 3D object detection API for a spin by training
+a CenterPillar model for Waymo's Open Dataset, which is a 3D object detection
+task for detecting cars and pedestrians for an autonomous vehicle.
+"""
+
+"""shell
+!pip install --upgrade git+https://github.com/keras-team/keras-cv
+!pip install tensorflow==2.11.0
+!pip install waymo-open-dataset-tf-2.11.0==1.5.1
+"""
+
+import keras_cv
+from keras_cv.callbacks import WaymoEvaluationCallback
+
+from keras_cv.datasets.waymo import load
+from keras_cv.datasets.waymo import transformer
+
+from keras_cv.layers.object_detection_3d import voxel_utils
+
+from keras_cv.layers import CenterNetLabelEncoder
+from keras_cv.layers import DynamicVoxelization
+
+from keras_cv.models.__internal__.unet import UNet
+
+from keras_cv.models.object_detection_3d.center_pillar import MultiClassDetectionHead
+from keras_cv.models.object_detection_3d.center_pillar import MultiClassHeatmapDecoder
+from keras_cv.models.object_detection_3d.center_pillar import MultiHeadCenterPillar
+
+import tensorflow as tf
+from tensorflow import keras
+
+"""
+3D object detection is the process of identifying, classifying,
+and localizing objects within a 3D space.  Inputs are often in the form of point
+clouds, although 2D images are sometimes used as inputs as well for multi-model
+3D object detection. KerasCV currently supports unimodal point cloud inputs for
+3D object detection.
+
+Point cloud inputs to 3D object detection models typically come from LIDAR
+sensors, and are generally structured as large batches of unsorted points
+captured at a single point in time.
+
+Before we get into the details of KerasCV's 3D object detection offerings, let's
+take a look at a pre-trained KerasCV 3D object detector. This preset has been
+trained on the entirety of the Waymo Open Dataset for 160 epochs, and achieves
+an average precision of 0.7 on the evaluation dataset for both vehicles and
+passengers.
+"""
+
+# TODO(ianstenbit): This doesn't work yet
+# pretrained_model = MultiHeadCenterPillar.from_preset("waymo_open_dataset_vehicles_and_pedestrians")
+# TODO(ianstenbit): Load up an eval frame and run predictions
+
+"""
+Great! Below we show a visualization of what some of these predictions look like
+alongside a rendering of the input point clouds.
+"""
+
+# TODO(ianstenbit): Include a link to a rendering of a visualization.
+
+"""
+Awesome! Now let's check out how to train this sort of model ourselves! First
+let's discuss how data is formatted for 3D object detection in KerasCV.
+In KerasCV, we adopt a data format where point clouds are represented as a
+dictionary with the following structure:
+
+```python
+point_cloud = {
+  "point_xyz": FloatTensor[batch_size, 3]
```

Comment:
I don't care a ton about this, but should we remove the `"point_"` prefix from all of these?  It seems a bit better without it.

Another Q: perhaps
{
    "coordinates"
}
instead of xyz?